torch/fx/experimental/symbolic_shapes.py

```diff
@@ -207,19 +207,31 @@ def free_symbols(val: Union[SymInt, torch.Tensor]) -> Set[sympy.Symbol]:
         return val.node.expr.free_symbols
     elif isinstance(val, sympy.Expr):
         return val.free_symbols
-    elif isinstance(val, (int, float, bool)):
+    elif isinstance(val, (int, float, bool, str)):
         return set()
     elif isinstance(val, torch.Tensor):
         return (
             free_symbols(val.size()) |
             free_symbols(val.stride()) |
             free_symbols(val.storage_offset())
         )
-    elif isinstance(val, (tuple, list)):
+    elif isinstance(val, (tuple, list, set)):
         r = set()
         for s in val:
             r |= free_symbols(s)
         return r
+    elif isinstance(val, dict):
+        r = set()
+        r |= free_symbols(set(val.keys()))
+        r |= free_symbols(list(val.values()))
```

Comment:
we shouldn't need to actually materialize the set/list here; maybe we just need to support `free_symbols` on an iterator.

I'm a bit suspicious about all of the new types supported here though, originally this basically let you cover FX node argument types, but now it is a bit more expansive than that.